project/app/main.py

```diff
@@ -1,20 +1,250 @@
+locs_df['admin_name'] = locs_df['admin_name'].apply(lowerify)
+
+states_map = {}
+# for each state, map their respective cities
+for state in list(locs_df.admin_name.unique()):
+    states_map[state] = locs_df[locs_df['admin_name']
+                                == state]['city_ascii'].to_list()
+
+# police brutality indentifying nlp
+model_path = os.path.join(os.path.dirname(
+    __file__), '..', 'model.pkl')
+model_file = open(model_path, 'rb')
+pipeline = pickle.load(model_file)
+model_file.close()
+
+# local csv backlog path
+backlog_path = os.path.join(os.path.dirname(
+    __file__), '..', 'backlog.csv'
+)
+
+# spacy nlp model
+nlp = spacy.load('en_core_web_sm')
+
+load_dotenv()
 
 app = FastAPI(
     title='Human Rights First Data Science API',
     description='Returns posts from Reddit\'s r/news subreddit on police brutality',
-    version='0.3',
+    version='0.4',
     docs_url='/',
 )
 
 app.include_router(predict.router)
 app.include_router(viz.router)
 app.include_router(getdata.router)
-app.include_router(update.router)
+app.include_router(ping.router)
+
+
+@app.on_event('startup')
+@repeat_every(seconds=60*60*24)  # 24 hours
+def run_update() -> None:
+    '''
+    Update backlog database with data from reddit.
+    '''
+    print('Updating backlog at %s' % datetime.now())
+
+    PRAW_CLIENT_ID = os.getenv('PRAW_CLIENT_ID')
+    PRAW_CLIENT_SECRET = os.getenv('PRAW_CLIENT_SECRET')
+    PRAW_USER_AGENT = os.getenv('PRAW_USER_AGENT')
+
+    reddit = praw.Reddit(
+        client_id=PRAW_CLIENT_ID,
+        client_secret=PRAW_CLIENT_SECRET,
+        user_agent=PRAW_USER_AGENT
+    )
+    # Grab data from reddit
+    data = []
+    for submission in reddit.subreddit("news").hot(limit=100):
+        data.append([
+            submission.id, submission.title, submission.url
+        ])
+    # construct a dataframe with the data
+    col_names = ['id', 'title', 'url']
+    df = pd.DataFrame(data, columns=col_names)
+
+    # pull the text from each article itself using newspaper3k
+    content_list = []
+    date_list = []
+    # go through each URL and use newspaper3k to extract data
+    for id_url in df['url']:
+        # use newspaper3k to extract text
+        article = Article(id_url)
+        article.download()
+        # if the article doesn't download, the error is thrown in parse()
+        try:
+            article.parse()
+        except:
+            # add null values to show no connection
+            content_list.append(None)
+            date_list.append(None)
+            continue
+        content_list.append(article.text)
+        # this will be null if newspaper3k can't find it
+        date_list.append(article.publish_date)
+    df['text'] = content_list
+    df['date'] = date_list
+
+    # drop any articles with missing data columns
+    df = df.dropna()
+    df = df.reset_index()
+    df = df.drop(columns='index')
+
+    # convert date column to pandas Timestamps
+    def timestampify(date):
```

Comment:
The use of nested functions should be pulled out and used inside of a utility file so that it may be used elsewhere in the project.